ykcompile/src/lib.rs

```diff
@@ -251,11 +257,77 @@ impl<TT> TraceCompiler<TT> {
                     ; mov [rdi + aoff], rax
                 );
             }
+            (Location::Register(reg), Location::Deref(boxed)) => match *boxed {
+                Location::Stack(off) => {
+                    dynasm!(self.asm
+                        ; mov Rq(reg), [rbp - off]
```

Comment:
Do I see it right that we dereference two things here?

Reply:
Once to get the pointer off the stack, again to dereference the pointer?

Reply:
Yes and no. We have a reference/pointer on the stack (in `[rbp-off]`). So we first move that pointer into a register (first instruction), and then dereference the pointer to get its value (second instruction).

Reply:
ok